crates/shared/src/sources/uniswap_v2/mod.rs

```diff
@@ -95,6 +92,16 @@ impl UniV2BaselineSourceParameters {
                 TESTNET_UNISWAP_INIT,
                 PoolReadingStyle::Default,
             )),
+            BS::Baoswap => {
+                return Some(Self {
+                    router: contracts::alloy::BaoswapRouter::deployment_address(
+                        &chain.parse::<u64>().expect("chain id should be an integer"),
```

Comment:
nit: given that we probably have to do the same thing for the other flavours we might as well already store the parsed chain id in a local variable.